File: Solutions/2134- Minimum Swaps to Group All 1's Together II/Solution.java
import java.util.*;

class Solution {
    public int minSwaps(int[] nums) {
        int countOne = 0;
        for (int num : nums) {
            if (num == 1) {
                countOne++;
            }
        }

        int n = nums.length;
        if (countOne == 0) {
            return 0;
        }

        int countZero = 0;
        for (int i = 0; i < countOne; i++) {
            if (nums[i] == 0) {
                countZero++;
            }
        }

        int answer = countZero;

        for (int i = 1; i < n; i++) {
            if (nums[i - 1] == 0) {
                countZero--;
            }
            
            if (nums[(i + countOne - 1) % n] == 0) {
                countZero++;
            }
            
            answer = Math.min(answer, countZero);
        }

        return answer;
    }

}

File: Solutions/2134- Minimum Swaps to Group All 1's Together II/Solution.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        count_one = nums.count(1)
        n = len(nums)
        
        if count_one == 0:
            return 0 

        answer = count_zero = nums[0:count_one].count(0)    

        for i in range(1, n):
            if nums[i - 1] == 0:
                count_zero -= 1
            
            if nums[(i + count_one - 1) % n] == 0:
                count_zero += 1
            
            answer = min(answer, count_zero)
        
        return answer
                
